app/react/src/client/preview/index.js

```diff
@@ -53,3 +59,5 @@ const renderUI = () => {
 };
 
 reduxStore.subscribe(renderUI);
+
+export const forceReRender = () => renderUI();
```

Comment:
If the story didn't change, why would this work?

Reply:
Never mind, I mis-interpreted the code.

Reply:
I was just looking at that code, funnily enough and I think it's a bit broken, I was going to send a PR to fix it. 

The current (psuedo) logic is:

```js
render() {
  // line linked above
  if (storyDifferent) {
      ReactDOM.unmountComponentAtNode(rootEl);
  }

  // later
  ReactDOM.render(element, rootEl);
}
```

So it *always* calls `render` and *sometimes* calls `unmount`. This actually breaks the React lifecycle in weird ways.

Reply:
So, if I was to fix the above (probably changing it to: `if (!storyDifferent) return;`, this PR would no longer work.

We should consider if we want to support this `forceRerender` functionality? I suppose we do? If so we will need a more declarative way to make it work, I don't think we should retain the above bug just to make it work.

Reply:
Probably the easiest way would be to add a "key" to the redux store that is changed when you call `forceRerender` and to also check if the key has changed.